library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java

```diff
@@ -971,27 +970,59 @@ private List<ElementSeed> getElementSeeds(final Iterable<Object> ids) {
             // Extract source and destination from ID list
             } else if (id instanceof Iterable) {
                 ((Iterable<?>) id).forEach(edgeIdList::add);
-            // Attempt to extract source and destination IDs from a string form of an array/list
-            } else if ((id instanceof String) && (((String) id).matches("^\\[.*,.*\\]$"))) {
+            // Attempt to extract source and destination IDs from a string form of an
+            // array/list
+            } else if ((id instanceof String) && (((String) id).matches("^\\[.*,.*\\]$")
+                || ((String) id).matches("^\\[.*,[a-zA-Z0-9]*,.*\\]$"))) {
                 edgeIdList = Arrays.asList(((String) id)
                         .replaceAll("\\s", "")
                         .replace("[", "")
                         .replace("]", "")
                         .split(","));
-            // Assume entity ID as fallback
+            // Assume entity ID as a fallback
             } else {
                 seeds.add(new EntitySeed(id));
             }
 
             // If found a list verify source and destination
             if (edgeIdList.size() == 2) {
                 seeds.add(new EdgeSeed(edgeIdList.get(0), edgeIdList.get(1)));
+            } else if (edgeIdList.size() == 3) {
+                seeds.add(new EdgeSeed(edgeIdList.get(0), edgeIdList.get(2)));
             }
         });
 
         return seeds;
     }
 
+    /**
+     * Determines the edge group used in the view based on supplied IDs.
+     * If ID contains a label, this is extracted and used in the view.
+     * If not, all edge groups are used.
+     *
+     * @param ids The iterable of IDs
+     * @return Set of edge labels for the view
+     */
+    private Set<String> getEdgeViewGroup(final Iterable<Object> ids) {
+        Set<String> labels = new HashSet<>();
+
+        ids.forEach(id -> {
+            if ((id instanceof String) && ((String) id).matches("^\\[.*,[a-zA-Z0-9]*,.*\\]$")) {
+                List<String> edgeIdList = Arrays.asList(((String) id)
+                        .replaceAll("\\s", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split(","));
```

Comment:
Same thing here regarding regex and capture groups
Can we reduce the duplication?